### PR TITLE
Onboard Migrate to Private Link Common Cmdlets

### PR DESCRIPTION
**Pull request by abhiroop-goyal**
## Description

Added PrivateLink support for ARM resources of the Migrate team.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 --->
 
 ## Upcoming Release
+* Onboard Migrate to Private Link Common Cmdlets
```
Comment (isra-fel) [inline src/Network/Network/ChangeLog.md:22]: Our practice is not to update Network's changelog for private link support, so please remove this.

File: src/Migrate/Changelog.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Onboard Migrate to Private Link Common Cmdlets
```
Comment (isra-fel) [inline src/Migrate/Changelog.md:21]: Migrate module does not change a thing, so change log should not be updated. Please remove this.

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi @abhiroop-goyal , please see my inline comments about removing change logs.

Besides, have you tested private link support after you registered resource types of migrate to network?

File: src/Network/Network/ChangeLog.md (hunk 1 above)
Comment (abhiroop-goyal) [reply src/Network/Network/ChangeLog.md]: Removed the changelog

File: src/Migrate/Changelog.md (hunk 2 above)
Comment (abhiroop-goyal) [reply src/Migrate/Changelog.md]: Removed the changelog

**Comment by abhiroop-goyal** [comment]
our integration with NRP is complete and we have tested it for all 3 of our resourceTypes.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**